pkgs/by-name/te/technitium-dns-server/package.nix

```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  makeWrapper,
+  dotnet-sdk_8,
+}:
+stdenvNoCC.mkDerivation rec {
+  pname = "technitium-dns-server";
+  version = "12.1";
+
+  src = fetchurl {
+    url = "https://download.technitium.com/dns/archive/${version}/DnsServerPortable.tar.gz";
+    hash = "sha256-G0M2xuYBZA3XXXaPs4pLrJmzAMbVJhiqISAvuCw3iZo=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,share/${pname}-${version}}
+    cp -r * $out/share/${pname}-${version}/.
+    rm $out/share/${pname}-${version}/start.{sh,bat}
+    rm $out/share/${pname}-${version}/DnsServerApp.exe
+    rm $out/share/${pname}-${version}/env-vars
+    rm $out/share/${pname}-${version}/systemd.service
```

Comment:
Should probably install this instead?

Reply:
Sorry, I don't understand what you mean?!

Reply:
You probably want to install the upstream service file to `$out/lib/systemd/system` if it works properly, so users can just add this package to `systemd.packages` and get the unit.

Reply:
Ah, OK. I'd like to opt for a module (later on). You need to open ports in the firewall for this to work properly and therefore, I'd much prefer a separate module where you can choose which ports to open (depending on the DNS protocols). Furthermore, the systemd.service doesn't work without patching. Overall, the separate module seems to be better.

Reply:
All right.

Reply:
Actually, maybe add a comment here explaining why the upstream unit is getting removed?

Reply:
Done.